Update csrankings.csv

SariSultan opened the pull request:
Fixed 194 errors (some appear more than once). The errors are of two types:

    Some links are not correct and gives error 404 not found. Those links have been changed to "NOSCHOLARPAGE".

    Some available links are updated by Google. In which they will give be redirected to a new link (using status 302). I updated this to the new links in case you are using crawling because it is extremely hard to automate 302 detection and updating it to redirection, where if you try to automate it it will give you a 404 error code instead of redirecting.

SariSultan committed:
Update csrankings.csv

Fixed 194 errors (some appear more than once). The errors are of two types:

    Some links are not correct and gives error 404 not found. Those links have been changed to "NOSCHOLARPAGE".

    Some available links are updated by Google. In which they will give be redirected to a new link (using status 302). I updated this to the new links in case you are using crawling because it is extremely hard to automate 302 detection and updating it to redirection, where if you try to automate it it will give you a 404 error code instead of redirecting.

emeryberger commented on csrankings.csv:
```diff
@@ -6280,7 +6280,7 @@ Jaelson Castro,UFPE,www.cin.ufpe.br/~jbc,Um1B6_4AAAAJ
 Jaesik Choi,UNIST,http://sail.unist.ac.kr,RqMLVzUAAAAJ
 Jaewoo Kang,Korea University,http://dmis.korea.ac.kr/people,iiIxp8cAAAAJ
-Jafar Habibi,Sharif University of Technology,http://sharif.ir/~jhabibi,fKvyreEAAAAJ
+Jafar Habibi,Sharif University of Technology,http://sharif.ir/~jhabibi,SCmyROQAAAAJ
```
I am spot checking these and randomly found this one, which is wrong (the current one is also wrong), so it should be replaced by NOSCHOLARPAGE.
https://scholar.google.com/citations?user=SCmyROQAAAAJ&hl=en

emeryberger commented on csrankings.csv:
```diff
@@ -7205,9 +7205,9 @@ John Williamson,University of Glasgow,http://www.dcs.gla.ac.uk/~jhw,fcvb8f8AAAAJ
-Johnny Wong,Iowa State University,http://www.cs.iastate.edu/people/johnny-wong,w3UeCAoAAAAJ
-Johnson Thomas,Oklahoma State University,http://www.cs.okstate.edu/~jpt,pyIZFggAAAAJ
+Johnny S. Wong,Iowa State University,http://www.cs.iastate.edu/people/johnny-wong,57pKMKwAAAAJ
+Johnny Wong,Iowa State University,http://www.cs.iastate.edu/people/johnny-wong,57pKMKwAAAAJ
```
Another randomly selected case where both are wrong: neither the original (w3UeCAoAAAAJ) nor the replacement (57pKMKwAAAAJ) are correct.

emeryberger commented on csrankings.csv:
```diff
@@ -9263,7 +9263,7 @@ Mario Lopez,University of Denver,http://portfolio.du.edu/mlopez,NOSCHOLARPAGE
 Mario Schaarschmidt,University of Koblenz-Landau,https://www.uni-koblenz-landau.de/de/koblenz/fb4/ifm/agschaarschmidt/team/schaarschmidt/mario-schaarschmidt,LmkFPYkAAAAJ
 Mario Vento,University of Salerno,http://mivia.unisa.it/people/vento,3PwXGpgAAAAJ
-Mariofanna Milanova,University of Arkansas - Little Rock,http://ualr.academia.edu/MariofannaMilanova,oJr0BfEAAAAJ
+Mariofanna Milanova,University of Arkansas - Little Rock,http://ualr.academia.edu/MariofannaMilanova,n4Uu1lIAAAAJ
```
This is my fourth random check and that's a 3/4 rate of incorrect replacement Google Scholar links. I'd be happier if you replaced these with NOSCHOLARPAGE.

emeryberger reviewed (changes requested):
I did a random check and found lots of issues...

SariSultan commented:
Yes, I did not check them manually. I relied on Google redirection for the original links which were not working. It seems that Google Scholar redirects to wrong URLs. 
About the links that do not work, they are 100% not working. 
I will double check the redirection links issue. 
Do you recommend a way to automating this process: for example we might compare the Google scholar page details (name and institute) with the available details in the csranking.csv, you think this will be good?

emeryberger commented:
I think it's tricky; I tried to search for Google Scholar links automatically (that's how they were initially populated) by searching for specific terms like name and institution (this no longer works because the scholarly package seems to not be working, or it's just Google blocking me). I'd rather substitute the NOSCHOLARPAGE than send to a dead link.

SariSultan committed:
Update csrankings.csv

SariSultan commented:
I have updated the redirected links to NOSCHOLARPAGE.

emeryberger commented:
I just checked a few and found lots of fine links. Please check your script to see what's going on. Also, it'd be better if you could submit PRs in smaller "bite-sized" portions to make them easier to thoroughly audit. Thanks.

SariSultan commented:
In the first comments I replaced the redirection links to correct ones:
For example,
Original link:: Dan Roth,University of Pennsylvania,http://l2r.cs.uiuc.edu,WHOHV3AAAAA **[NOT WORKINGJ]**

Updated on _1st commit_: 
Dan Roth,University of Pennsylvania,http://l2r.cs.uiuc.edu,E-bpPWgAAAAJ **[Correct]**

One the second commit I changed all redirection links to NOSCHOLARPAGE, because we did not have a method to verify if Google was redirecting to the correct page or not. I think it is better to make a commit with a fix to broken links (not redirection) until i find a way to automate redirection checking.

emeryberger commented:
Too much to go through manually.
